### ISSUE TITLE
The validation image is black when displayed in the tensorboard

### ISSUE THREAD

**Issue by todochenxi** [huggingface/diffusers#9558]
### Describe the bug

when I use the script `train_text_to_image_lora_sdxl.py` to train my lora, the validation image in tensorboard is black ....  just like this
![image](https://github.com/user-attachments/assets/a2e2cd87-9eb3-47e8-b2a0-5caff9fcf9f1)

### Reproduction

the code in train_text_to_image_lora_sdxl
```pythonfor tracker in accelerator.trackers:
        phase_name = "test" if is_final_validation else "validation"
        if tracker.name == "tensorboard":
            np_images = np.stack([np.asarray(img) for img in images])
            tracker.writer.add_images(phase_name, np_images, epoch, dataformats="NHWC")
        if tracker.name == "wandb":
            tracker.log(
                {
                    phase_name: [
                        wandb.Image(image, caption=f"{i}: {args.validation_prompt}") for i, image in enumerate(images)
                    ]
                }
            )
```            

### System Info

diffusers-cli env
WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.3.1+cu121 with CUDA 1201 (you have 2.4.1+cu121)
    Python  3.12.4 (you have 3.12.3)
  Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
  Memory-efficient attention, SwiGLU, sparse and more won't be available.
  Set XFORMERS_MORE_DETAILS=1 for more details

Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- 🤗 Diffusers version: 0.31.0.dev0
- Platform: Linux-6.8.0-40-generic-x86_64-with-glibc2.35
- Running on Google Colab?: No
- Python version: 3.12.3
- PyTorch version (GPU?): 2.4.1+cu121 (True)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Huggingface_hub version: 0.25.1
- Transformers version: 4.38.0
- Accelerate version: 0.34.2
- PEFT version: 0.7.0
- Bitsandbytes version: 0.44.0
- Safetensors version: 0.4.5
- xFormers version: 0.0.27
- Accelerator: NVIDIA GeForce RTX 3090, 24576 MiB
NVIDIA GeForce RTX 3090, 24576 MiB
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by todochenxi** [comment]
@yiyixuxu @sayakpaul @DN6

**Comment by asomoza** [comment]
Hi, did you test the lora if it trained successfully? If you don't have NaNs in the training loss, probably the black images its because you're using the default VAE with fp16 which produces black images most of the time.

**Comment by jeongiin** [comment]
Hi, I encountered a similar issue in `train_dreambooth_lora_flux.py`.
I was able to resolve it using this method.


https://github.com/huggingface/diffusers/blob/8e7d6c03a366fdb0f551ce7b92f0871c863d4e08/examples/dreambooth/train_dreambooth_lora_sdxl.py#L212-L215

to
~~~python
autocast_ctx = torch.autocast(accelerator.device.type, dtype=torch_dtype)
~~~


If it works for you as well, I would like to submit a PR for this issue.
I hope it helps!

**Comment by todochenxi** [comment]
> Hi, did you test the lora if it trained successfully? If you don't have NaNs in the training loss, probably the black images are because you're using the default VAE with fp16 which produces black images most of the time.

Thanks for your reply, lora is able to reason normally, I saw the description of vae in the xl readme, but I used dxlUnstableDiffusers_v11 to train my lora, maybe it's the vae, I'd be willing to do the training again, thanks ：）

**Comment by todochenxi** [comment]
> Hi, I encountered a similar issue in `train_dreambooth_lora_flux.py`. I was able to resolve it using this method.
> 
> https://github.com/huggingface/diffusers/blob/8e7d6c03a366fdb0f551ce7b92f0871c863d4e08/examples/text_to_image/train_text_to_image_lora_sdxl.py#L142-L145
> 
> to
> 
> ```python
> autocast_ctx = torch.autocast(accelerator.device.type, dtype=torch_dtype)
> ```
> 
> If it works for you as well, I would like to submit a PR for this issue. I hope it helps!

Thanks for the advice, I will train again, btw, the sks dog is cute ^v^

**Comment by todochenxi** [comment]
> > Hi, I encountered a similar issue in `train_dreambooth_lora_flux.py`. I was able to resolve it using this method.
> > https://github.com/huggingface/diffusers/blob/8e7d6c03a366fdb0f551ce7b92f0871c863d4e08/examples/text_to_image/train_text_to_image_lora_sdxl.py#L142-L145
> > 
> > to
> > ```python
> > autocast_ctx = torch.autocast(accelerator.device.type, dtype=torch_dtype)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If it works for you as well, I would like to submit a PR for this issue. I hope it helps!
> 
> Thanks for the advice, I will train again, btw, the sks dog is cute ^v^

And [](https://github.com/bghira/SimpleTuner/blob/main/documentation/quickstart/FLUX.md ) Here's a workflow for reasoning that I hope will be useful to you

**Comment by todochenxi** [comment]
> > Hi, did you test the lora if it trained successfully? If you don't have NaNs in the training loss, probably the black images are because you're using the default VAE with fp16 which produces black images most of the time.
> 
> Thanks for your reply, lora is able to reason normally, I saw the description of vae in the xl readme, but I used dxlUnstableDiffusers_v11 to train my lora, maybe it's the vae, I'd be willing to do the training again, thanks ：）

Yes, it works, thanks
